### ISSUE TITLE
CI_REGISTRY has local-registry prefix

### ISSUE THREAD

**Issue by avoidik** [firecow/gitlab-ci-local#1119]
**Minimal .gitlab-ci.yml illustrating the issue**

```yaml
# .gitlab-ci-local/variables.yml
---
global:
  CI_REGISTRY: "valid.url.here:1234"
```

```yaml
# .gitlab-ci.yml
cool_job_here:
  stage: build
  image: ${CI_REGISTRY}/some.path.here/docker:cli
  services:
    - alias: docker
      name: ${CI_REGISTRY}/some.path.here/docker:dind
  script:
    - whoami
```

```terminal
$ export GCL_FILE='.gitlab-ci.yml' 
$ gcl cool_job_here
parsing and downloads finished in 35 ms
cool_job_here starting local-registry.valid.url.here/some.path.here/docker:cli (build)
cool_job_here > still running...
cool_job_here copied to docker volumes in 17 s
Error: Command failed with exit code 1: docker pull local-registry.valid.url.here/some.path.here/docker:dind
Error response from daemon: Get "https://local-registry.valid.url.here/v2/": dial tcp: lookup local-registry.valid.url.here on 127.0.0.53:53: no such host
    at makeError (/Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Job.pullImage (/Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/src/job.ts:784:13)
    at /Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/src/job.ts:456:21
    at async Promise.all (index 0)
    at Job.start (/Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/src/job.ts:453:13)
    at /Users/user.name/.nvm/versions/node/v18.16.0/lib/node_modules/gitlab-ci-local/node_modules/p-map/index.js:57:22
```

**Expected behavior**

I would like to be able to control `CI_REGISTRY` variable so that no prefixes will be added

**Host information**

```terminal
$ gcl --version
4.46.1
```

**Containerd binary**

```terminal
$ docker --version
Docker version 25.0.3, build 4debf411d1
```

**Additional context**

culprit:
https://github.com/firecow/gitlab-ci-local/commit/5ecafe0ba32fe5bdf71392dcdda13698efd1b098#diff-a66bffdb979d130e04df61edb944cc04d2fdedf8117ca2732cb6d119d5f5e7b2R136

**Comment by avoidik** [comment]
it seems `CI_REGISTRY` is just ignored if I set it as a global variable

**Comment by avoidik** [comment]
disregard, I should've been set `--home "$(cwd)"`

**Comment by peterbbeu** [comment]
@firecow is it possible to make this configurable (whether or not `local-` prefix should be set)? I'm happy to make  PR for this. The reason is that we use the actual gitlab registry in our test pipelines; right now we are always just overriding the CI_REGISTRY manually to remove the 'local-' bit. But that always gives the warning that the CI_* vars should not be overwritten.

**Comment by firecow** [comment]
Are you thinking about an option like this `--registry http://gitlab.local` ?

**Comment by avoidik** [comment]
What about rephrasing the warning? In my opinion it's totally okay to override these variables the way it's described in the OP.  If look closer, do we really need this to be a warning? Just show a normal info log with a list of overridden variables. In the end it's me controlling the app, not the app controlling me.

**Comment by firecow** [comment]
I would be fine with removing the warning, and just printing a list of variables that's been overridden, so we stand a chance to debug issues where logs are pasted.

**Comment by peterbbeu** [comment]
> Are you thinking about an option like this `--registry http://gitlab.local` ?

For us, the use case is that we use `registry.gitlab.com` so we end up with this in `.gitlab-ci-local-variables.yml` (and a corresponding warning, which actually think is a good thing):

```yaml
CI_REGISTRY: registry.gitlab.com
```

Whereas otherwise we get `local-registry.gitlab.com`. So in our case, something like `--disable-local-registry-prefix` would work just fine.

I believe it bases this off the repo url right? Which in our case is something like

```
git@gitlab.com:gitlab-group/project.git
```

**Comment by avoidik** [comment]
@peterbbeu if I remember correctly, the prefix is added only when the local variables file is in use, will it work for you if you try the global variables file?

technically, everything except local should work:

https://github.com/firecow/gitlab-ci-local/blob/2163e7e7c07cec77219848d2bb937a10575f3ae6/src/job.ts#L167

**Comment by peterbbeu** [comment]
@avoidik I think it's hardcoded: https://github.com/firecow/gitlab-ci-local/blob/2163e7e7c07cec77219848d2bb937a10575f3ae6/src/job.ts#L332

And basically what I am proposing is an option to change this from `local-registry.` to just `registry.` (to mimic regular gitlab behaviour)

**Comment by avoidik** [comment]
Yes, it's a hardcoded **predefined** variable, and it's still possible to override all these variables by using:
- global variables
- job variables
- matrix variables
- file variables
- arg variables

**Comment by peterbbeu** [comment]
Yes I know, but as it stands if you override it (in order to achieve default gitlab behaviour), it generates a warning. Which is not a disaster I suppose but I'm not sure if it's desirable.

Perhaps I'm misunderstanding you, do you mean it's possible to override it yet not generate a warning?

**Comment by ANGkeith** [comment]
Yes, the warning can be ignored. 

For context, the intention of the warning is just re-iterating the "warning" from gitlab.com docs. 

<img width="1274" height="832" alt="Image" src="https://github.com/user-attachments/assets/bd2aa807-528c-4ce7-babd-588b22c3145e" />

I am not sure if it can be better phrased and also, from the docs it's not explicitly stated which predefined variables can cause unexpected behavior


---
In fact, by doing the following:
> we use registry.gitlab.com so we end up with this in .gitlab-ci-local-variables.yml

it's kinda a HACK... 

i think?(i might be wrong) the "actual"/"intended" gitlab.com way of updating `CI_REGISTRY` is through the [gitlab.rb](https://gitlab.com/gitlab-org/omnibus-gitlab/blob/c9d421c2fe63e1ceb77649a8cb41f699ac3a2b88/files/gitlab-config-template/gitlab.rb.template#L1030) rather than overriding the predefined variable.

**Comment by peterbbeu** [comment]
> it's kinda a HACK...

How exactly is that a hack? We use hosted gitlab.com, and if we don't change ANYTHING at all, our `CI_REGISTRY` is `registry.gitlab.com`. The prefixing of `CI_REGISTRY` with `local-` is non-standard behaviour, that we override (to standard gitlab.com behaviour!) in our local pipelines to conform with what we expect in the actual gitlab.com instance. 

Perhaps you misunderstand that I'm doing this ONLY in `gitlab-ci-local` and NOT in gitlab.com itself, in order to achieve the same behaviour that you get in gitlab.com.

Here's a simple PoC confirming that: https://gitlab.com/petervv/gitlab-ci-test/-/jobs/12192993787#L25 (I configured nothing else in that)

I mean as I said before, I'm fine to ignore that warning, but I still think it's strange to have a warning when all you're doing is trying to get closer to default gitlab.com behaviour.

**Comment by ANGkeith** [comment]
yes, i mean doing this in `gitlab-ci-local` , in my opinion is a "HACK"/"Workaround"

**Comment by peterbbeu** [comment]
I suppose, but as it stands, there is no other way to mimic gitlab.com behaviour in this way.
